File: app/src/main/java/me/dara/memoapp/ui/view/DatePickerAppCompat.java
package me.dara.memoapp.ui.view;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDialogFragment;
import java.util.Calendar;
import me.dara.memoapp.R;

/**
 * @author sardor
 */
public class DatePickerAppCompat extends AppCompatDialogFragment {
  public DatePickerDialog.OnDateSetListener listener;

  @NonNull @Override public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

    Calendar c = Calendar.getInstance();
    int year = c.get(Calendar.YEAR);
    int month = c.get(Calendar.MONTH);
    int day = c.get(Calendar.DAY_OF_MONTH);
    return new DatePickerDialog(requireContext(), R.style.DialogThemeStyle, listener, year, month,
        day);
  }
}

File: app/src/main/java/me/dara/memoapp/ui/view/DatePickerAppCompat.kt
package me.dara.memoapp.ui.view

import android.app.DatePickerDialog
import android.app.Dialog
import android.os.Bundle
import android.widget.DatePicker
import androidx.appcompat.app.AppCompatDialogFragment
import me.dara.memoapp.R

import java.util.*

/**
 * @author sardor
 */
class DatePickerAppCompat : AppCompatDialogFragment(), DatePickerDialog.OnDateSetListener {


  lateinit var listener: (view: DatePicker?, year: Int, month: Int, dayOfMonth: Int) -> Unit

  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    val c = Calendar.getInstance()
    val year = c.get(Calendar.YEAR)
    val month = c.get(Calendar.MONTH)
    val day = c.get(Calendar.DAY_OF_MONTH)

    return DatePickerDialog(requireContext(), R.style.DialogThemeStyle,this, year, month, day)
  }

  override fun onDateSet(view: DatePicker?, year: Int, month: Int, dayOfMonth: Int) {
    listener(view, year, month, dayOfMonth)
  }

}